hepdata/modules/records/utils/records_update_utils.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+import math
+
+from hepdata.modules.records.utils.doi_minter import generate_dois_for_submission
+from hepdata.modules.submission.api import get_latest_hepsubmission
+from hepdata.modules.records.utils.common import get_record_by_id
+from hepdata.modules.records.utils.workflow import update_record
+from hepdata.modules.inspire_api.views import get_inspire_record_information
+from hepdata.ext.elasticsearch.api import index_record_ids
+from hepdata.modules.email.api import notify_publication_update
+from hepdata.resilient_requests import resilient_requests
+
+
+RECORDS_PER_PAGE = 10
+
+
+def update_record_info(inspire_id, recid='', send_email=False, verbose=False):
+    inspire_id = inspire_id.replace("ins", "")
+
+    if recid == '':
+        hep_submission = get_latest_hepsubmission(inspire_id=inspire_id)
+        if hep_submission is None:
+            print("Failed to retrieve hep submission for {0}".format(inspire_id))
+            return
+        recid = hep_submission.publication_recid
+
+    if verbose:
+        print("Updating recid {} with information from inspire record {}".format(recid, inspire_id))
+
+    updated_record_information, status = get_inspire_record_information(inspire_id)
+
+    if status == 'success':
+        record_information = get_record_by_id(recid)
+        same_information = True
+        for key, value in updated_record_information.items():
+            if key not in record_information or record_information[key] != value:
+                same_information = False
+                break
+        if verbose:
+            print('Information needs to be updated:', not(same_information))
+        if same_information:
+            return
+        record_information = update_record(recid, updated_record_information)
+    else:
+        print("Failed to retrieve publication information for {0}".format(inspire_id))
+        return
+
+    if hep_submission.overall_status == 'finished':
+        index_record_ids([record_information["recid"]])
+        generate_dois_for_submission.delay(inspire_id=inspire_id)  # update metadata stored in DataCite
+        if send_email:
+            notify_publication_update(hep_submission, record_information)   # send email to all participants
+
+
+def get_inspire_records_updated_since(date, verbose=False):
+    """Returns all inspire records updated since YYYY-MM-DD or #int as number of days since today (1 = yesterday)"""
+    return _get_inspire_records_updated('since', date, verbose)
+
+
+def get_inspire_records_updated_on(date, verbose=False):
+    """Returns all inspire records updated on YYYY-MM-DD or #int as number of days since today (1 = yesterday)"""
+    return _get_inspire_records_updated('on', date, verbose)
+
+
+def _get_inspire_records_updated(on_or_since, date, verbose=False):
+
+    specified_time = _get_time(date)
+
+    if verbose:
```

Comment:
Should we be using the Logger class here and DEBUG or INFO, rather than a custom `verbose` option?

Reply:
I agree, so I've replaced the `verbose` option with `log.info`.  I realised that there's no easy way to change the log level (by default, `WARNING`), so I introduced a variable `LOGGING_CONSOLE_LEVEL = 'WARNING'` (used by [invenio-logging](https://invenio-logging.readthedocs.io/en/latest/configuration.html#invenio_logging.config.LOGGING_CONSOLE_LEVEL)) in `config.py`.  A more verbose option of `WARNING` or `DEBUG` can then be set in `config_local.py`.  I modified all modules that use `logging` to set the log level using `LOGGING_CONSOLE_LEVEL`.

Reply:
I've now removed `LOGGING_CONSOLE_LEVEL` and instead recommended `export FLASK_ENV=development` in the installation instructions (and in `docker-compose.yml`).  I've also improved `update_record_info` to return a status instead of `None`, which makes the test `test_update_record_info` in `tests/records_test.py` more meaningful.  Please take another look, @alisonrclarke.